initial commit

NekhoroshkinaElena opened the pull request:
проверить прохождение тестов

NekhoroshkinaElena committed:
initial commit

NekhoroshkinaElena committed:
feat: создать новые классы для работы с базой данных FilmDbStorage, GenreDbStorage, MPADbStorage, UserDbStorage
refactor: перенести методы по добавленю и удалению лайков и получению топ фильмов из FilmService в FilmDbStorage
refactor: перенести методы добавления, удаления и получения общих друзей из UserService в UserDbStorage
delete: удалить InMemoryFilmStorage.java и InMemoryUserStorage.java
feat: добавить классы GenreNotFoundException и MpaNotFoundException для обработки ошибок рейтинга и жанров
feat: добавить документацию к FilmStorage, UserStorage, GenreStorage, MPAStorage
feat: обавить классы ValidateFilm, ValidateRatingMpa, ValidateUser для проверки данных

NekhoroshkinaElena commented:
Добрый день, добавила в приложение новую функциональность, очень жду вашей оценки:)

chewmag commented on src/main/java/ru/yandex/practicum/filmorate/storage/impl/FilmDbStorage.java:
```diff
@@ -0,0 +1,199 @@
+            Pair mpa = new Pair<>(rowSet.getInt("id_mpa"), rowSet.getString("rating"));
+            Pair genre1 = new Pair(rowSet.getInt("id_genre"), rowSet.getString("genre"));
+            if (genre1.name == null || genre1.id == null) {
+                Film film = new Film(
```
В этом же классе есть метод makeFilm, он как раз создает Film из ResultSet'a

chewmag commented on src/main/java/ru/yandex/practicum/filmorate/storage/impl/UserDbStorage.java:
```diff
@@ -0,0 +1,132 @@
+        jdbcTemplate.update(sql, currentId, user.getName(), user.getLogin(),
+                user.getBirthday(), user.getEmail());
+        user.setId(currentId);
+        getUniqueID();
```
А почему не передать getUniqueID в user.setId ?

chewmag reviewed (changes requested):
Приветствую Елена!

По заданию требовалось добавить новые имплементации к интерфейсам FilmStorage, UserStorage, но хранение в памяти кануло в лету ) Отсюда бы потребовалось использовать Qualifier, чтобы подключить нужную имплементацию. Еще один момент, который нужно исправить: при получении фильмов в методе FilmDbStorage:getFilmById нужно собрать жанры из уже выполненного запроса, а не делать отдельный новый.

NekhoroshkinaElena committed:
fix: сделать переменные в классе Pair приватными
fix: убрать аннотацию @Qualifier
fix: заменить if на else в методе UpdateFilm
fix: исправить метод makeFilm что бы создавать объект фильма из одного запроса
fix: передать getUniqueID в user.setId и getUniqueID в film.setId
fix: исправить метод makeUser что бы не дублировать код
style: исправить регистр при запросе к базам данных

NekhoroshkinaElena commented:
Добрый день, постаралась исправить указанные ошибки. Очень долго не могла разобраться с ошибкой: "В методе FilmDbStorage:getFilmById нужно собрать жанры из уже выполненного запроса, а не делать отдельный новый." Надеюсь, что правильно вас поняла:)

chewmag commented on src/main/java/ru/yandex/practicum/filmorate/storage/impl/UserDbStorage.java:
```diff
@@ -0,0 +1,121 @@
+                "u.id " +
+                "FROM friendship f " +
+                "INNER JOIN users u ON f.friend_id = u.id " +
+                "WHERE (f.user_id = ? OR f.user_id = ?) AND (f.friend_id != ? AND f.friend_id != ?)";
```
достаточно один раз f.user_id = ?  и f.friend_id != ? , в таком виде это дублирование )

chewmag reviewed (approved):
Приветствую Елена!

Улучшения в FilmDbStorage реализованы, можно мерджить PR в main 👍

NekhoroshkinaElena committed:
fix: убрать дублирование кода в методе getListCommonFriends в классе UserDbStorage

NekhoroshkinaElena commented:
убрала прямое дублирование кода
